### PR TITLE
Object Window Niceties 

### LINKED ISSUE #67249: Performacne of the alt click / turf stat panel is now extremely terrible
## Reproduction:

Alt-clicking something / pulling up the "turf" stat panel now seems to lag your game to absurd levels.

### PR DESCRIPTION
**Pull request by LemonInTheDark**
## About The Pull Request

Alright. I got bored and polished up the object/alt click window.

It had a few issues:
First, we generated all our images in bulk, as soon as requested
Second, the caching was global, despite only working on a client to client basis
Third, we only generated up to 10 images. This could be fine, but the javascript code will continuiously rerender assuming unrendered images will come eventually, and they well, weren't. This caused MASSIVE clientside lag
Fourth and finally, I did not like how moving away from the viewed turf lagged behind, in sync with the stat tab update. Looked bad.

I've resolved all these.
I solved the first three issues by reworking how obj images were generatated and managed.

Rather then storing a basic cache on the subsystem, and doing all the image generation at once, we queue up image generation as we like, and generate images inside a new processing subsystem fire.
This isn't the best solution, since it still eats cpu somewhat, but it's a whole lot better then the other options, outside either removing the need to getflat, or somehow predicting what items a client will want to see

I've started storing three bits of info. 
First, a list of all the objects we currently want to display.
Second, a list of atom -> image html
Third, a list of atoms to imageify.

This information is stored on a datum on /client, since I want this to have a lifetime linked to well, clients.

I've used this datum to solve that fourth bit, using a component I made for parallax a bit back. This lets me react to our client's mob, and update the tab linked to that, rather then on a subsystem call by call basis.

That's about it.

## Why It's Good For The Game

fixes https://github.com/tgstation/tgstation/issues/67249

Kills lag, smoother response, kills client lag.

## Changelog

:cl:
fix: The alt click tab (obj tab?) will no longer cause massive clientside lag (unless you alt click that 2000 obj large tofu pile)
fix: The alt click menu is snappier now, try usin it and let me know if anything blows up yeah?
/:cl:

### CODE REVIEW

**Comment by Mothblocks** [comment]
fixes #67249?

**Comment by LemonInTheDark** [comment]
Yes

File: code/controllers/subsystem/statpanel.dm [hunk 1]
```diff
@@ -276,9 +298,107 @@ SUBSYSTEM_DEF(statpanels)
...
 /client/var/datum/tgui_window/stat_panel
+
+/// Datum that holds and tracks info about a client's object window
+/// Really only exists because I want to be able to do logic with signals
+/// And need a safe place to do the registration
+/datum/object_window_info
+	/// list of atoms to show to our client via the object tab, at least currently
+	var/list/atoms_to_show = list()
+	/// list of atom -> image string for objects we have had in the right click tab
+	/// this is our cachign
```
Comment (san7890) [inline code/controllers/subsystem/statpanel.dm:311]:
```suggestion
	/// this is our caching
```

**Review by san7890 — COMMENTED** [review]
smally small spell check :)

**Pull request merged**